Content.Server/Fluids/EntitySystems/PuddleSystem.Spillable.cs

```diff
@@ -117,7 +117,9 @@ private void OnGotEquipped(Entity<SpillableComponent> entity, ref GotEquippedEve
             return;
 
         // block access to the solution while worn
-        AddComp<BlockSolutionAccessComponent>(entity);
+        var blockerComp = AddComp<BlockSolutionAccessComponent>(entity);
+        blockerComp.Solution = entity.Comp.SolutionName;
+        Dirty(entity, blockerComp);
 
```

Comment:
Isn't this going to get bulldozed if somehow they wear 2 things with this?

Reply:
no it will throw because AddComp

Reply:
Argh, good points.

It's easy enough to switch to EnsureComp, but then there's the bulldozing issue. I'll swap the single value for a set.

Reply:
Wait, no. The component is added to the clothing entity and blocks access to its own solutions, not the wearer's.  Someone wearing multiple things with Spillable won't cause any problems, they'll just each block their own internal solution.

Reply:
I suppose there could be an issue if multiple systems tried to use BlockSolutionAccess on one entity. Spillable is the only thing that uses it now, but maybe something else would in the future. Hm.

Reply:
Does it work if you make blockerComp.Solution a list?